libraries/search/lib/Archive.ts

```diff
@@ -106,8 +106,8 @@ export class Archive<T extends Encoding> {
     // Do not update if the encoding is already assigned to the objective function
     if (oldEncoding && oldEncoding === encoding) {
       Archive.LOGGER.debug("encoding already assigned to objective function");
-      throw new Error(
-        shouldNeverHappen("encoding already assigned to objective function")
+      throw new AlreadyAssignedError(
```

Comment:
It would be better to make these errors more specific (e.g., EncodingAlreadyAssignedError)